### PR TITLE
Override the plugin URL found in update-center.json with JENKINS_PLUGINS_URL env var value

### PR DESCRIPTION
**Pull request by martinda**
I would like to propose a feature that lets users override the URL found in the `update-center.json` file with a URL provided by the environment variable `JENKINS_PLUGINS_URL`.

This feature is useful in the following cases:
* The mirrors are down or broken
* You want faster downloads from an intranet server
* You have a policy for caching the plugins on an intranet server

Combining this feature with `--jenkins-update-center <custom_url>` lets you download plugins from servers you specify.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ if the user doesn't have a home directory when it will go to: `$(pwd)/.cache/jen
 * `JENKINS_UC_DOWNLOAD`: used to configure the URL from where plugins will be downloaded from. Often used to cache or to proxy the Jenkins plugin download site. 
 If set then all plugins will be downloaded through that URL.
   
+* `JENKINS_PLUGINS_URL`: Rewrites the plugin url to the specified value before a plugin is downloaded. Use this option to download from a specific server and bypass the mirrors and force the use of your own server. For example, if your server has the Jenkins plugins at `https://company.com/jenkins-plugins` then simply set this variable to that value. From the specified URL, the plugin must be found under `pluginName/version/pluginName.hpii`. There is no need to re-write the `update-center.json` file as this feature will re-write the URL right before the plugin is fetched. This is useful when the mirrors are broken or if you have an internal policy requirement. You still need access to the official `update-center.json` on the Jenkins servers file.
```
Comment (oleg-nenashev) [inline README.md:75]:
```suggestion
* `JENKINS_PLUGINS_URL`: Rewrites the plugin url to the specified value before a plugin is downloaded. Use this option to download from a specific server and bypass the mirrors and force the use of your own server. For example, if your server has the Jenkins plugins at `https://company.com/jenkins-plugins` then simply set this variable to that value. From the specified URL, the plugin must be found under `pluginName/version/pluginName.hpi`. There is no need to re-write the `update-center.json` file as this feature will re-write the URL right before the plugin is fetched. This is useful when the mirrors are broken or if you have an internal policy requirement. You still need access to the official `update-center.json` in the Jenkins upda center.
```

**Review by oleg-nenashev — COMMENTED** [review]
No strong opinion. The only difference from `JENKINS_UC_DOWNLOAD` is the layout format (the original option expects '/plugins' in the path).

File: README.md (hunk 1 above)
Comment (martinda) [reply README.md:75]: @oleg-nenashev I think `JENKINS_UC_DOWNLOAD` requires the `update-center.json` to be hosted on a server as well. `JENKINS_PLUGINS_URL` does not. So a user could continue to fetch the update-center.json from the official server while at the same time use another server for the plugins. I do not know if this is a common occurence, but it is certainly mine.

A use case might clarify:
```
export JENKINS_PLUGINS_URL=https://server.com/customPath
java -jar jenkins-plugin-manager.jar \
    --plugin-file plugins.yml \
    --plugin-download-directory plugins \
    --jenkins-version 2.235.3
```

And thanks for the quick review!

Comment (timja) [reply README.md]: Jenkins uc download shouldn’t require that, the point of it is to just override the download url

**Review by timja — CHANGES_REQUESTED** [review]
I can't see any reason for this, the only difference between the two options is whether /plugins should be in the URL or not and that's not what your change justification is

I've replied on gitter to your questions which should fix your issue

File: README.md (hunk 1 above)
Comment (martinda) [reply README.md]: Ah, you're right, I see what I did. I messed up the documentation.

**Comment by martinda** [comment]
I would like to present PR #278 instead. The explanation I provide in this one is wrong and I was not able to rebase it cleanly.

**Pull request closed without merging**